src/incidents/view/ViewIncidentDetails.tsx

```diff
@@ -22,27 +24,42 @@ function ViewIncidentDetails(props: Props) {
   const history = useHistory()
   const { t } = useTranslator()
   const { data, isLoading } = useIncident(incidentId)
-  const { data: patient } = usePatient(data?.patient)
-  const [mutate] = useResolveIncident()
+  const [incident, setIncident] = useState<Incident | undefined>(data)
 
-  if (data === undefined || isLoading) {
+  const [mutate] = useUpdateIncident()
+
+  useEffect(() => {
+    setIncident(data)
+  }, [data, isLoading])
+
+  if (incident === undefined || isLoading) {
     return <Spinner type="DotLoader" loading />
   }
 
   const onResolve = async () => {
-    await mutate(data)
+    await mutate({ ...incident, status: 'resolved' })
+    history.push('/incidents')
+  }
+  const onUpdate = async () => {
+    await mutate({ ...incident, status: 'reported' })
     history.push('/incidents')
   }
-
   const getButtons = () => {
     const buttons: React.ReactNode[] = []
-    if (data.status === 'resolved') {
+    if (incident.status === 'resolved') {
       return buttons
     }
-
     if (permissions.includes(Permissions.ResolveIncident)) {
       buttons.push(
-        <Button onClick={onResolve} color="primary" key="incidents.reports.resolve">
+        <Button className="mr-2" color="success" onClick={onUpdate} key="incidents.reports.update">
+          {t('incidents.reports.update')}
+        </Button>,
+        <Button
+          onClick={onResolve}
+          className="mr-2"
+          color="primary"
+          key="incidents.reports.resolve"
+        >
           {t('incidents.reports.resolve')}
```

Comment:
To clarify, is the button here supposed to say resolve or update? (in the translations, since you added an "update" field)

Reply:
ah, let me double check!

Reply:
whoops, i left both buttons. one for "update" which lets user edit most fields, and resolve which just updates the status to resolved.